escalate/core/views/experiment.py

```diff
@@ -241,42 +209,13 @@ def post(self, request, *args, **kwargs):
                             else:
                                 setattr(query, field, data['value'])
                                 query.save()
-
+                                
                 if template_name in SUPPORTED_CREATE_WFS:
                     lsr_edoc = Edocument.objects.get(ref_edocument_uuid=exp_template.uuid, title='LSR file')
                     if template_name == 'liquid_solid_extraction':
-                        # q3 contains concentraiton logic
-                        if any([f.has_changed() for f in q3_formset]):
-                            data = {}  # Stick form data into this dict
-                            for i, form in enumerate(q3_formset):
-                                if form.is_valid():
-                                    query = q3[i]
-                                    data[query.parameter_def_description] = form.cleaned_data['value'].value
-
-                            hcl_vols, h2o_vols = hcl_mix(data['stock_concentration'],
-                                                         data['total_vol'],
-                                                         np.fromstring(data['hcl_concentrations'].strip(']['), sep=',')
-                                                         )
-                            related_exp = 'workflow__experiment_workflow_workflow__experiment'
-                            h2o_dispense_action_set = WorkflowActionSet.objects.get(**{f'{related_exp}': experiment_copy_uuid,
-                                                                                       'description__contains': 'H2O'})
-                            hcl_dispense_action_set = WorkflowActionSet.objects.get(**{f'{related_exp}': experiment_copy_uuid,
-                                                                                       'description__contains': 'HCl'})
-                            update_dispense_action_set(h2o_dispense_action_set, h2o_vols)
-                            update_dispense_action_set(hcl_dispense_action_set, hcl_vols)
-                            new_lsr_pk, lsr_msg = update_lsr_edoc(lsr_edoc,
-                                                                  experiment_copy_uuid,
-                                                                  exp_name,
-                                                                  vol_hcl=list(hcl_vols*1000),
-                                                                  vol_h2o=list(h2o_vols*1000))
+                        new_lsr_pk, lsr_msg = liquid_solid_extraction(q3_formset,q3,experiment_copy_uuid,lsr_edoc,exp_name)
```

Comment:
I really like what youve done with the supported_wfs() function and breaking each piece template specific logic into its own method. We may be able to go even further with the abstraction s.t. we do not have to add a method call to this file, but rather just do something like 

```
expBuilder = ExperimentBuilder('wf_name')
expBuilder.build()
context.extend(exp_builder.response_data)
```

But we can slowly iterate towards something that general, I think we have other things to fix first before we attempt that level of generality.